week3/sum3.py

```diff
@@ -0,0 +1,122 @@
+from sets import Set
+import numpy as np
+
+#Create unit test cases for python 
+array = None #test case 1  
+array1 = [] #test case 2
+array2 = [0,1] #test case 3
+array3 = [0,1,-1] #test case 4
+array4 = [0,1,1] #test case 5
+array5 = [2,0,-1,-2,1] #test case 6 
+array6 = [0,0,0,0,0] #test case 7
+array7 = [3,-1,-1,-1,0,-3,2, -6,6,-3] #test case 8
+
+Arrays =[array, array1, array2, array3, array4, array5, array6, array7]
+Num_arrays_in_Arrays = len(Arrays) 
+
+#Define Functions
+
+#cehck input function
+def Check_Input(current_array):
+    if current_array is None or current_array == []:
+  		return False 
+    if not(all(isinstance(element, int) for element in current_array)):
+    	return False
+    return True 
+
+#get length of current array
+def Current_Array_Length(current_array):
+	length = len(current_array)
+	return length
+
+#Find all the possibilities for 3 numbers sum it up to 0 in current_array
+def Get_Unique_Triplets(ArrayLength, current_array):
+    
+	#Special case for only 3 elements in the array
+    if(ArrayLength == 3): 
+        SumArray = sum(current_array)
+        if(SumArray == 0):
+            sortArray = sorted(current_array)
+            return(sortArray)
+        else:
+            return None
+	
+	#Special case for all the elements are the same in the array
+    if(all(current_number == 0 for current_number in current_array)):
+        return([0,0,0])
+    
+	#Special case if all numbers are positive or negative
```

Comment:
I sort of understand that why it is not good to have this function, since we can't guarantee the input. 
How about if I just check the first element and the last element to see if both of them are 0 ? Because I already sorted the array first, the checking above will make sure that this whole array only contains 0.